scarb/src/core/manifest/toml_manifest.rs

```diff
@@ -409,20 +412,28 @@ impl TomlManifest {
             PackageId::new(name, version, source_id)
         };
 
-        let mut dependencies = Vec::new();
-        for (name, toml_dep) in self.dependencies.iter().flatten() {
+        let mut dependencies =
+            Vec::with_capacity(self.dependencies.iter().len() + self.dev_dependencies.iter().len());
```

Comment:
```suggestion
        let mut dependencies = Vec::new();
```

Nitpick: a negligible micro-optimization (this vector won't grow beyond 100ish items in sensible projects), which hinders readability a bit ;)